src/OpenXR/Silk.NET.OpenXR/Structs/GraphicsBinding.cs

```diff
@@ -0,0 +1,100 @@
+    public struct GraphicsBinding
+    {
+        [FieldOffset(0)] public StructureType Type;
+
+        [FieldOffset(0)] public GraphicsBindingD3D11KHR D3D11Khr;
+        [FieldOffset(0)] public GraphicsBindingD3D12KHR D3D12Khr;
+        [FieldOffset(0)] public GraphicsBindingEGLMNDX EglMndx;
+        [FieldOffset(0)] public GraphicsBindingOpenGLESAndroidKHR OpenGLESAndroidKhr;
+        [FieldOffset(0)] public GraphicsBindingOpenGLWaylandKHR OpenGLWaylandKhr;
+        [FieldOffset(0)] public GraphicsBindingOpenGLWin32KHR OpenGLWin32Khr;
+        [FieldOffset(0)] public GraphicsBindingOpenGLXcbKHR OpenGLXcbKhr;
+        [FieldOffset(0)] public GraphicsBindingOpenGLXlibKHR OpenGLXlibKhr;
+        [FieldOffset(0)] public GraphicsBindingVulkan2KHR Vulkan2Khr;
+        [FieldOffset(0)] public GraphicsBindingVulkanKHR VulkanKhr;
+        
+        public static implicit operator GraphicsBinding(GraphicsBindingD3D11KHR binding)
+            => Unsafe.As<GraphicsBindingD3D11KHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingD3D12KHR binding)
+            => Unsafe.As<GraphicsBindingD3D12KHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingEGLMNDX binding)
+            => Unsafe.As<GraphicsBindingEGLMNDX, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingOpenGLESAndroidKHR binding)
+            => Unsafe.As<GraphicsBindingOpenGLESAndroidKHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingOpenGLWaylandKHR binding)
+            => Unsafe.As<GraphicsBindingOpenGLWaylandKHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingOpenGLWin32KHR binding)
+            => Unsafe.As<GraphicsBindingOpenGLWin32KHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingOpenGLXcbKHR binding)
+            => Unsafe.As<GraphicsBindingOpenGLXcbKHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingOpenGLXlibKHR binding)
+            => Unsafe.As<GraphicsBindingOpenGLXlibKHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingVulkan2KHR binding)
+            => Unsafe.As<GraphicsBindingVulkan2KHR, GraphicsBinding>(ref binding);
+
+        public static implicit operator GraphicsBinding(GraphicsBindingVulkanKHR binding)
+            => Unsafe.As<GraphicsBindingVulkanKHR, GraphicsBinding>(ref binding);
+
+        private readonly T Cast<T>(StructureType assertType)
+        {
+            if (Type != assertType)
+            {
+                static void Throw() => throw new InvalidCastException();
+                Throw();
+            }
+
+            return Unsafe.As<GraphicsBinding, T>(ref Unsafe.AsRef(in this));
+        }
```

Comment:
What is this?!

Reply:
Per the documentation, GraphicsBinding is a union to address the fact there are so many GraphicsBinding structures, and to allow for abstractions.

Per the code, Cast is a utility function used by the explicit operators where we want to cast it back to an explicit type so we can check the structure type as well to check that it's a valid cast.